deployment/recipes/roles/set-defaults/tasks/main.yml

```diff
@@ -0,0 +1,60 @@
+### We have a single role to set default variables used throughtout other roles, playbooks, etc.
+### The reason for this is that many settings need to agree, despite being used in different roles
+### and we want to ensure that the defaults are set/updated in a single place. This also serves to
+### gather the configurable parameters in a single place for easier referece.
+---
+- name: Set default values
+  set_fact:
+    ## enable verbose debugging tasks
+    verbose_debug_tasks: "{{ verbose_debug_tasks | default(False) }}"
+    ## some standard ansible vars where we want to ensure assignment and defaults
+    ansible_user: "{{ ansible_user | default(ansible_user_id) }}"
+    ansible_python_interpreter: "{{ ansible_python_interpreter | default('{{discovered_interpreter_python}}') }}"
+    ## extra system-level pacakges to install
+    extra_system_packages: "{{ extra_system_packages | default([]) }}"
+    ## our custom ansible module has dependencies, install them to a venv
+    venv_for_ansible: "{{ venv_for_ansible | default('{{ ansible_env.HOME }}/ansible_venv') }}"
+    ## variables related to the version of volttron to be installed
+    volttron_use_local_source: "{{ volttron_use_local_source | default(False) }}"
```

Comment:
if this is True how do you set the local source?  or will this be a path if wanting to use local source?

Reply:
I saw later that it uses volttron_root.  Perhaps a comment to where that's set in the host-config.

Reply:
It is currently defined to be two dirs up relative to the path of the playbook. Could make it 3 options:
false -> use github
true (not a string) -> current logic, relative to playbook
true (is string) -> string is path